torch/_inductor/codegen/common.py

```diff
@@ -953,6 +953,47 @@ def ctx():
 
         return ctx()
 
+    def rethrow_exception(self):
+        @contextlib.contextmanager
+        def ctx():
+            self.writelines(
+                [
+                    "std::atomic_flag err_flag = ATOMIC_FLAG_INIT;",
+                    "std::exception_ptr eptr;",
```

Comment:
May I know why are we adding c++ specific code in the common buffer class?

Reply:
Thanks for advice, I moved it to `cpp_utils.py`.